server/src/main/java/org/elasticsearch/health/node/UpdateHealthInfoCacheAction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.node;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.health.node.action.TransportHealthNodeAction;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+
+public class UpdateHealthInfoCacheAction extends ActionType<AcknowledgedResponse> {
+
+    public static class Request extends ActionRequest {
+        private final String nodeId;
+        private final DiskHealthInfo diskHealthInfo;
```

Comment:
@gmarouli and I have discussed this outside of this PR, but it's probably worth mentioning the tradeoffs of putting a disk-specific field here (and I don't think this discussion is worth blocking this PR for, just starting it here). If we do this then that means that whenever we add a new indicator that uses the health node we'll need to modify this transport action and the one that fetches the data (which doesn't exist yet). I think it also means we'll need to modify HealthService for each one so that it knows which data to send to which indicator.
An alternative is to use a generic list or map and use reflection when serializing/deserializing the data. But that has some unpleasant tradeoffs as well.
So I think for now we're proposing that we'll have:

- Concrete fields in UpdateHealthInfoCacheAction and the upcoming FetchHealthInfoCacheAction for each health node data type
- Concrete fields in LocalHealthMonitor for each health node data type
- HealthService will keep health node indicators separate, and explicitly send the correct data to each health node indicator (or something like that).

And if we wind up with a large number of indicators that use the health node we'll revisit this because at that point the downsides of reflection might be worth it.

Reply:
> HealthService will keep health node indicators separate, and explicitly send the correct data to each health node indicator (or something like that).

We could also pass all the health info to the indicators and each indicator will pick the information it needs, that might help with keeping the health service as clean as possible.

One more option that we could do to keep things generic but without introducing reflection, is that we keep a generic class `HealthInfo` that will contain the `HealthStatus` and the `Cause` and we use this for all indicators. Then we have a map as shown follows:
```
Map.of(
    "disk", new HealthInfo(HealthStatus.YELLOW, Cause.ABOVE_HIGH_WATERMARK)
    "threadpool", new HealthInfo(HealthStatus.GREEN, null)
    );
```

I would like to avoid reflection at all cost... But I am open to using strings as labels :).